src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -761,7 +764,10 @@ private void ApplyJournalStateAfterFlush(CancellationToken token,
                         }
                     }, byteStringContext);
 
-                    edi?.Throw();
+                    if (edi != null)
+                        edi.Throw();
+                    else if (appliedSuccessfully == false)
+                        throw new InvalidOperationException($"Journal state was not applied successfully after the flush (waited - {sp.Elapsed}, last flushed tx: id - {record.TransactionId}, written to journal - {record.WrittenToJournalNumber})");
```

Comment:
In what scenario can _this_ happern?

Reply:
Please check the tests that have been added - those are two scenarios when this can happen. It's easier to revert the fix and debug those tests to see the issue than describing it. It was very complex to narrow it down.